### PR TITLE
Using mutate_rows() instead of row.commit()

### PR DESCRIPTION
**Pull request by sumit-ql**
This is a fix for https://github.com/googleapis/google-cloud-python-happybase/issues/4

### CODE REVIEW

File: unit_tests/test_batch.py [hunk 1]
```diff
@@ -82,6 +82,7 @@ def test_constructor_with_non_default_wal(self):
 
     def test_constructor_with_non_positive_batch_size(self):
         table = object()
+
```
Comment (tseaver) [inline unit_tests/test_batch.py]: Please remove.

File: unit_tests/test_batch.py [hunk 2]
```diff
@@ -91,32 +92,33 @@ def test_constructor_with_non_positive_batch_size(self):
 
     def test_constructor_with_batch_size_and_transactional(self):
         table = object()
+
```
Comment (tseaver) [inline unit_tests/test_batch.py]: Please remove.

File: unit_tests/test_batch.py [hunk 3]
```diff
@@ -91,32 +92,33 @@ def test_constructor_with_non_positive_batch_size(self):
...
         transaction = True
         with self.assertRaises(TypeError):
             self._make_one(
                 table, batch_size=batch_size, transaction=transaction)
 
     def test_send(self):
-        table = object()
+        low_level_table = _MockLowLevelTable()
+        table = _MockTable(low_level_table)
+
```
Comment (tseaver) [inline unit_tests/test_batch.py]: Please remove the blank line.

File: unit_tests/test_batch.py [hunk 4]
```diff
@@ -91,32 +92,33 @@ def test_constructor_with_non_positive_batch_size(self):
...
         self.assertNotEqual(batch._mutation_count, 0)
         self.assertNotEqual(row_map, {})
+        self.assertEqual(len(table._low_level_table.rows_mutate), 0)
 
         batch.send()
         self.assertEqual(row_map.clear_count, 1)
-        self.assertEqual(row1.commits, 1)
-        self.assertEqual(row2.commits, 1)
         self.assertEqual(batch._mutation_count, 0)
+        self.assertEqual(len(table._low_level_table.rows_mutate), 2)
```
Comment (tseaver) [inline unit_tests/test_batch.py]: Needs an assertion that the contents of `rows_mutate` match expected values.

File: unit_tests/test_batch.py (hunk 1 above)
Comment (sumit-ql) [reply unit_tests/test_batch.py]: Done.

File: unit_tests/test_batch.py (hunk 2 above)
Comment (sumit-ql) [reply unit_tests/test_batch.py]: Done.

Comment (sumit-ql) [reply unit_tests/test_batch.py]: Removed blank line.

File: unit_tests/test_batch.py (hunk 3 above)
Comment (sumit-ql) [reply unit_tests/test_batch.py]: Done.

File: unit_tests/test_batch.py (hunk 4 above)
Comment (sumit-ql) [reply unit_tests/test_batch.py]: Added assertion to check expected values. Thanks.

**Comment by sumit-ql** [comment]
@tseaver Incorporated review comments, please review.

**Pull request merged**